### PR TITLE
plug some more misc memory leaks

### PR DESCRIPTION
**Pull request by ggouaillardet**

### CODE REVIEW

**Comment by artpol84** [comment]
@ggouaillardet you may want to rebase this now

File: src/client/pmix_client.c [hunk 1]
```diff
@@ -236,6 +236,24 @@ static void evhandler_reg_callbk(pmix_status_t status,
...
+    PMIX_DESTRUCT(&pmix_client_globals.myserver);
+    *active = false;
+}
+
+
+static void pmix_destruct_my_server(void)
+{
+    pmix_event_t ev;
+    volatile bool active = true;
+    event_assign(&ev, pmix_globals.evbase, -1,
```
Comment (artpol84) [inline src/client/pmix_client.c]: We have a macro for that:
https://github.com/pmix/master/blob/master/src/client/pmix_client_get.c#L179

**Comment by ggouaillardet** [comment]
@artpol84 done, thanks for the pointer

**Comment by artpol84** [comment]
I'll do the review later. Hopefully tomorrow.
Thank you for the nice job!

**Pull request merged**